lib/purserl.ex

```diff
@@ -1769,25 +1770,32 @@ defmodule DevHelpers.Purserl do
     end
   end
 
-  defp ioputs(device \\ :stdio, item) do
-    IO.puts(device, item)
+  defp await_task(t) do
+    case Process.alive?(t) do
+      true ->
+        # IO.puts("Waiting for Erlang compilation to finish ...")
+        ref = Process.monitor(t)
+        receive do
+          {:DOWN, ^ref, :process, ^t, :normal} -> :ok
+        end
+      false ->
+        :ok
+    end
+  end
```

Comment:
why are you reimplementing https://hexdocs.pm/elixir/Task.html#async/1 ? Are you sure you didn't miss an edge-case?

Reply:
This and `async` are not the same. The main difference is that `async` sends (usually) two messages to the calling process once the async task is completed, and handling those two messages are not really necessary for purerlex, we only care that the process exited normally. These processes are still linked using `spawn_link` so if any of them crashes purerlex will crash as well. There are ways of using `async` instead, but the usecase for `async` is running async code in an otherwise sync function. Purerlex is mostly async already, so using `async` is somewhat redundant. 

Did I miss an edge case? Probably, but I'm not certain. I think the only way for this to go wrong is if the processes are tampered with. I looked at the elixir source and I could probably use `Task.start_link` instead of `spawn_link`, but I'm unsure what the practical difference is. Might make this change later.

Reply:
Thanks for explaining! You did in fact consider Task.async, and you made an informed decision. That's all I need, really. If you add a comment in cases like this (e.g. `-- NOTE[emattfolk]: Considered Task.async, but this sends fewer messages so should be faster.`), I won't even have to ask :)

Reply:
It is very hard to know what comments are relevant when writing the code. But I'll try to keep in mind the descisions I make along the way